### DISCUSSION TITLE
How to retain SNPs that match to data from multiple samples

### DISCUSSION THREAD

**Discussion by diegojehu** [PGScatalog/pgsc_calc#216] [Q&A]
Hi, 

I'm aiming to evaluate 3 custom PRSs in 3 different sets of individuals (i.e. samples from three different biobanks). To allow appropriate comparison across sets, I want to make sure the exact same SNPs are used to calculate the individuals' scores. However, I suspect SNP matching will vary across these datasets (due to different genotyping chips and different imputation steps) potentially resulting in slightly different numbers of SNPs used to calculate the scores in dataset A vs dataset B vs dataset C. 

I wanted to ask if there were any suggestions on how to add an additional SNP matching step so that only the SNPs that are matched across datasets are retained to compute scores in all datasets.

Any suggestions are welcome! :)

Thanks

**Comment by nebfield** [comment]
Thanks for your interesting question! I think it requires some manual work. Assuming your three biobanks are held in separate environments, you could try something like:

* Export variant information from each biobank (e.g. plink2 pvar files)
* Calculate the intersection of variants (you could try `plink2 --extract-intersect` or do something custom)
* Import the intersection of variants and make a subset of your biobanks with `plink2 --extract <intersection> --make-pgen`

Another option would be to run `match_variants` from [our utilities package](https://github.com/PGScatalog/pgscatalog_utils) for each biobank and take common row numbers to make a new scoring file. However, the utilities package documentation isn't great and we're working on improving it (we mostly use these programs internally in the calculator). 

Also, some people make scores with a "reference standardised framework". There's [a new preprint that describes this approach across 5 biobanks](https://www.medrxiv.org/content/10.1101/2023.11.20.23298215v1). Perhaps you might want to try a similar approach?
